### PR TITLE
Add deploy cycle breakdown app

### PR DESCRIPTION
**Pull request by snabar**
App provides breakdown of pr review time, merge time, build time, deploy time, for every deployment in a time window. PR review time and merge time calculations need to be updated once approvals are available in the VCS schema.

### CODE REVIEW

File: scripts/requirements.txt [hunk 1]
```diff
@@ -3,3 +3,4 @@ jinja2==2.11.2
 jsonpath_rw_ext
 requests==2.24.*
 ruamel.yaml
+dataclasses-json
```
Comment (cjwooo) [inline scripts/requirements.txt:6]: Why is this needed here?

File: apps/dora-metrics/deploy-cycle-breakdown/app.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    deploy_time: int
+
+    # TODO: Update pr_review_time and pr_merge_time calculations once approvals are available
+    # TODO: Should we be adding null checks for every field here, like the one for the mergedPullRequest?
+    def __init__(self, deployment):
+        self.id = deployment["id"]
+        self.application = deployment["application"]["name"]
+        self.ended_at = int(deployment["endedAt"])
+        self.pr_review_time = None
+        self.pr_merge_time = int(deployment["build"]["commit"]["mergedPullRequest"]["mergedAt"]) - \
```
Comment (cjwooo) [inline apps/dora-metrics/deploy-cycle-breakdown/app.py:25]: Perhaps make `deployment["build"]["commit"]["mergedPullRequest"]` its own var to make this line more readable

File: scripts/requirements.txt (hunk 1 above)
Comment (snabar) [reply scripts/requirements.txt:6]: to serialize the data class for the response ... is there a better way to do this conversion?

Comment (cjwooo) [reply scripts/requirements.txt:6]: Oh I see it's required to run it the tests

**Review by cjwooo — APPROVED** [review]
